### PR TITLE
Allow user define heading id.

### PR DESCRIPTION
**Pull request by vwxyzh**
Allow user define heading id.
For example:
```md
# <a id="x"></a>Title
```
will generate following html:
```html
<h1 id="x">Title</h1>
```

### CODE REVIEW

File: src/Microsoft.DocAsCode.Dfm/Aggregators/HeadingIdAggregator.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+namespace Microsoft.DocAsCode.Dfm
+{
+    using System.Text.RegularExpressions;
+
+    using Microsoft.DocAsCode.MarkdownLite;
+
+    public class HeadingIdAggregator : MarkdownTokenAggregator<MarkdownHeadingBlockToken>
+    {
+        private static readonly Regex OpenARegex = new Regex(@"^\<a +(?:name|id)=\""([\w \-\.]+)\"" *\>$", RegexOptions.Compiled);
```
Comment (yufeih) [inline src/Microsoft.DocAsCode.Dfm/Aggregators/HeadingIdAggregator.cs]: <a id="x" class="xliff"></a>

Comment (yufeih) [inline src/Microsoft.DocAsCode.Dfm/Aggregators/HeadingIdAggregator.cs]: `<a id="x" class="xliff"></a>`

**Comment by hellosnow** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Dfm/Aggregators/HeadingIdAggregator.cs (hunk 1 above)
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Dfm/Aggregators/HeadingIdAggregator.cs]: Do not apply id in such heading, keep out put like following:
```html
<h1 id="{autogenid}">
<a id="x" class="xliff"></a>Title
</h1>
```

**Pull request merged**